index.html

```diff
@@ -1108,9 +1461,80 @@ <h2>Reporting of Third-party Subresource Fetch Failures</h2>
+      similarly restricted to <a>trustworthy origins</a>. This disallows a
+      transient HTTP MITM from trivially abusing NEL as a persistent
+      tracker.
+      </p>
+
+      <p>
+      NEL is intended to augment existing server-side monitoring.  NEL reports
+      should only be sent to the owner of the service being requested.  For
+      errors that occur during <a>DNS resolution</a>, the user agent MUST ensure
+      that the <a>NEL policy</a> was received from the owner of the <a>domain
+        namespace tree</a> that contains the <a>policy origin</a>.  For errors
+      that occur during <a>secure connection establishment</a> or
+      <a>transmission of request and response</a>, the user agent MUST ensure
+      that the <a>NEL policy</a> was received from the owner of the
+      <a>server</a> that the <a>request</a> was sent to.
+      </p>
 
-      <p>To mitigate some of the above risks, NEL registration is restricted to <a>trustworthy origins</a>, and delivery of network error reports is similarly restricted to <a>trustworthy origins</a>. This disallows a transient HTTP MITM from trivially abusing NEL as a persistent tracker.</p>
+      <p>
+      This rationale explains the treatment of the <a>received IP address</a>
+      and <a>subdomains</a> flag of a <a>NEL policy</a>.  By checking that the
+      policy's <a>received IP address</a> matches the IP address of the
+      <a>server</a>, NEL extends the trust boundary of the policy to include not
+      just the policy's <a data-lt="policy origin">origin</a>, but also the
+      specific server that the user agent is communicating with.  This helps
+      prevent (for instance) DNS rebinding attacks, where an attacker delivers a
+      long-lived <a>NEL policy</a> from a server that they own, and then changes
+      their name servers to resolve the <a>policy origin</a> to a server they
+      don't control.  Without the <a>received IP address</a> verification, this
+      would cause user agents to send reports about the second server to the
+      attacker.
+      </p>
+
+      <p>
+      Similarly, <a data-lt="subdomains">subdomain</a> <a>NEL policies</a> are
+      limited, and can only be used to generate reports about subdomains of the
+      <a>policy origin</a> during the <a>DNS resolution</a> phase of a
+      <a>request</a>.  During this phase, there is no <a>server</a> to verify
+      ownership of, and the fact that the policy was received from a superdomain
+      of the <a>request</a>'s <a>origin</a> is enough to establish ownership of
+      the error.  This allows the owners of a particular portion of the
+      <a>domain namespace tree</a> to use NEL to detect <a
+        href="#dns-misconfiguration"></a> errors, while preventing them from
+      using malicious DNS entries to collect information about servers they
+      don't control.
+      </p>
+
+      <p>
+      To prevent information leakage, NEL reports about a <a>request</a> MUST

```

Comment:
Beware "MUST"s in security and privacy implications. Since they don't appear in the algorithms that have to actually make them happen, it's easy for implementers to miss them. That said, I don't have a good guideline for what to do instead.

Reply:
All of these statements should be implied by the algorithms earlier in the document, so I changed the wording so that this doesn't imply that it's adding additional requirements, just elaborating on the rationale for the existing ones.